src/ui/public/filter_editor/filter_editor.html

```diff
@@ -6,110 +6,115 @@
       filter
     </div>
 
-    <div
+    <button
       class="kuiModalHeaderCloseButton kuiIcon fa-times"
       ng-click="filterEditor.onCancel()"
-    ></div>
+      aria-label="Close filter popover"
+    ></button>
   </div>
 
   <div class="kuiModalBody">
-    <div ng-show="filterEditor.showQueryDslEditor()">
-      <filter-query-dsl-editor
-        filter="filterEditor.filter"
-        on-change="filterEditor.setQueryDsl(queryDsl)"
-      ></filter-query-dsl-editor>
-      <div
-        ng-show="filterEditor.isEditingQueryDsl"
-        ng-click="filterEditor.isEditingQueryDsl = false"
-      >
-        <button class="kuiMenuButton kuiMenuButton--basic">
-          <span class="kuiMenuButton__icon kuiIcon fa-chevron-left"></span>
-          <span>Back</span>
-        </button>
-      </div>
-    </div>
+    <!-- Filter definition -->
+    <div class="kuiVerticalRhythm">
+      <div class="kuiVerticalRhythmSmall filterEditor__labelBar">
+        <label class="kuiLabel">
+          Filter
+        </label>
 
-    <div
-      class="kuiFieldGroup kuiVerticalRhythm kuiFieldGroup--alignTop"
-      ng-hide="filterEditor.showQueryDslEditor()"
-    >
-      <div class="kuiFieldGroupSection">
-        <filter-field-select
-          index-patterns="filterEditor.indexPatterns"
-          field="filterEditor.field"
-          on-select="filterEditor.onFieldSelect(field)"
-        ></filter-field-select>
-      </div>
+        <div>
+          <a
+            class="kuiLink"
+            ng-click="filterEditor.isEditingQueryDsl = !filterEditor.isEditingQueryDsl"
+            kbn-accessible-click
+          >
+            <span ng-if="filterEditor.isEditingQueryDsl">
+              Search query values
+            </span>
 
-      <div class="kuiFieldGroupSection">
-        <filter-operator-select
-          ng-if="filterEditor.field"
-          field="filterEditor.field"
-          operator="filterEditor.operator"
-          on-select="filterEditor.onOperatorSelect(operator)"
-        ></filter-operator-select>
+            <span ng-if="!filterEditor.isEditingQueryDsl">
+              Edit raw query
+            </span>
+          </a>
+        </div>
       </div>
 
-      <div class="kuiFieldGroupSection kuiFieldGroupSection--wide filterEditor__wideField">
-        <filter-params-editor
-          ng-if="filterEditor.field && filterEditor.operator"
-          field="filterEditor.field"
-          operator="filterEditor.operator"
-          params="filterEditor.params"
-        ></filter-params-editor>
-      </div>
-    </div>
+      <!-- Filter dropdowns -->
+      <div
+        class="kuiFieldGroup kuiVerticalRhythmSmall kuiFieldGroup--alignTop"
+        ng-if="!filterEditor.showQueryDslEditor()"
```

Comment:
Answered my own question. `ng-if` adds an isolate scope which makes it difficult to access controller properties.